src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/DoNotGuardSetAddOrRemoveByContainsTests.cs

```diff
@@ -0,0 +1,1159 @@
+
+        public MyClass()
+        {{
+            if ({(method == "Add" ? "!" : string.Empty)}[|MySet.Contains(""Item"")|])
+                MySet.{method}(""Item"");
+        }}" + CSNamespaceAndClassEnd;
+
+            string fixedSource = CSUsings + CSNamespaceAndClassStart + $@"
+        private readonly {interfaceType} MySet = {SetCreationExpression(concreteType)}
+
+        public MyClass()
+        {{
+            MySet.{method}(""Item"");
+        }}" + CSNamespaceAndClassEnd;
+
+            await VerifyCS.VerifyCodeFixAsync(source, fixedSource);
+        }
+
+        [Theory]
+        [InlineData("ImmutableHashSet<string>", "Add")]
+        [InlineData("ImmutableHashSet<string>", "Remove")]
+        [InlineData("ImmutableSortedSet<string>", "Add")]
+        [InlineData("ImmutableSortedSet<string>", "Remove")]
+        public async Task SupportsSetWithAddOrRemoveReturningGenericType_ReportsDiagnostic_CS(string setType, string method)
+        {
+            string source = CSUsings + CSNamespaceAndClassStart + $@"
+        private {setType} MySet = {setType[..setType.IndexOf('<', StringComparison.Ordinal)]}.Create<string>();
+
+        public MyClass()
+        {{
+            if ({(method == "Add" ? "!" : string.Empty)}[|MySet.Contains(""Item"")|])
+               MySet = MySet.{method}(""Item"");
+        }}" + CSNamespaceAndClassEnd;
+
+            await VerifyCS.VerifyCodeFixAsync(source, source);
+        }
+
+        [Theory]
+        [InlineData("IImmutableSet<string>", "ImmutableHashSet<string>", "Add")]
+        [InlineData("IImmutableSet<string>", "ImmutableHashSet<string>", "Remove")]
+        [InlineData("IImmutableSet<string>", "ImmutableSortedSet<string>", "Add")]
+        [InlineData("IImmutableSet<string>", "ImmutableSortedSet<string>", "Remove")]
+        public async Task SupportsSetWithAddOrRemoveReturningGenericTypeWithInterfaceType_ReportsDiagnostic_CS(string interfaceType, string concreteType, string method)
+        {
+            string source = CSUsings + CSNamespaceAndClassStart + $@"
+        private {interfaceType} MySet = {concreteType[..concreteType.IndexOf('<', StringComparison.Ordinal)]}.Create<string>();
+
+        public MyClass()
+        {{
+            if ({(method == "Add" ? "!" : string.Empty)}[|MySet.Contains(""Item"")|])
+               MySet = MySet.{method}(""Item"");
+        }}" + CSNamespaceAndClassEnd;
+
+            await VerifyCS.VerifyCodeFixAsync(source, source);
+        }
+        #endregion
+
+        #region Helpers
+        private const string CSUsings = @"using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;";
+
+        private const string CSNamespaceAndClassStart = @"namespace Testopolis
+{
+    public class MyClass
+    {
+";
+
+        private const string CSNamespaceAndClassEnd = @"
+    }
+}";
+
+        private const string VBUsings = @"Imports System
+Imports System.Collections.Generic
+Imports System.Collections.Immutable";
```

Comment:
💡 It would be preferable to inline all of these

Reply:
Just to be sure that I understand correctly:

Change this:

```C#
[Fact]
public async Task AddWithVariableAssignment_ReportsDiagnostic_CS()
{
    string source = CSUsings + CSNamespaceAndClassStart + @"
private readonly HashSet<string> MySet = new HashSet<string>();

public MyClass()
{
    if (![|MySet.Contains(""Item"")|])
    {
        bool result = MySet.Add(""Item"");
    }
}
" + CSNamespaceAndClassEnd;

    await VerifyCS.VerifyCodeFixAsync(source, source);
}
```

To:

```C#
[Fact]
public async Task AddWithVariableAssignment_ReportsDiagnostic_CS()
{
    string source = """
        using System.Collections.Generic;

        class C
        {
            private readonly HashSet<string> MySet = new HashSet<string>();

            void M()
            {
                if (![|MySet.Contains("Item")|])
                {
                    bool result = MySet.Add("Item");
                }
            }
        }
        """;

    await VerifyCS.VerifyCodeFixAsync(source, source);
}
```

Reply:
> Just to be sure that I understand correctly:

Correct, it meant to move `CSNamespaceAndClassStart` and `CSNamespaceAndClassEnd` inline.

Reply:
Changed in [6f69703](https://github.com/dotnet/roslyn-analyzers/pull/6767/commits/6f6970341a2f77e921f92ec32e6d050fa7bf780c).